utils.js

```diff
@@ -15,6 +15,67 @@ const resolveFileAndLine = (file, classname, output) => {
     return { filename, line: parseInt(line) };
 };
 
+const appendStdOutStdErrAnnotations = async (file, testsuite, pathFailuresSet, annotations) => {
+    // gradle saves stdout/stderr directly to the XML file
+    const stdOutData = ((testsuite["system-out"] && testsuite["system-out"]._cdata) ||
+                       (testsuite["system-out"] && testsuite["system-out"]._text) || '').trim();
+    const stdErrData = ((testsuite["system-err"] && testsuite["system-err"]._cdata) ||
+                        (testsuite["system-err"] && testsuite["system-err"]._text) || '').trim();
+    let raw_details;
+    if (!stdErrData && !stdErrData) {
+        // maven saves stdout/stderr to an independent file, remove "TEST-" from the file name.
+        // http://maven.apache.org/surefire/maven-surefire-plugin/test-mojo.html#redirectTestOutputToFile
+        let fileExtIndex = file.lastIndexOf(".");
+        if (fileExtIndex < 0) {
+            fileExtIndex = file.length;
+        }
+        // windows path has been canonicalized to '/' path separators by resolvePath. -1 is OK
+        // because we add 1 to look for testFilePrefix.
+        const lastSeparatorIndex = file.lastIndexOf("/");
+        const testFilePrefix = "TEST-";
+        let stdOutFile;
+        if (file.substr(lastSeparatorIndex + 1, testFilePrefix.length) == testFilePrefix) {
+            stdOutFile = file.substr(0, lastSeparatorIndex + 1) +
+                file.substring(lastSeparatorIndex + testFilePrefix.length + 1, fileExtIndex);
+        } else {
+            stdOutFile = file.substring(0, fileExtIndex);
+        }
+        stdOutFile += "-output.txt";
+
+        let stdOutErrData;
+        try {
+            stdOutErrData = await fs.promises.readFile(stdOutFile);
+            core.debug(`Found stdout/stderr in ${stdOutFile}`);
+        } catch (err) {
+            stdOutErrData = null;
+        }
+        raw_details = stdOutErrData ? ("stdout/stderr:\n" + stdOutErrData) : null;
+    } else {
+        core.debug(`Found stdout/stderr in ${file}`);
+        raw_details = ("stdout:\n" + stdOutData + "\n\n\n" + "stderr:\n" + stdErrData);
+    }
+
+    if (raw_details) {
+        // stdout/stderr is captured at the testsuite or test file level, but the metadata within
```

Comment:
I'm not sure of a scenario where this can happen, but wanted to open up for discussion in case anyone has more info.